archive/backup.go

```diff
@@ -149,18 +165,8 @@ func processExportStream(
 	logger hclog.Logger,
 	writer io.Writer,
 	targetFrom, targetTo uint64,
-) (*uint64, *uint64, error) {
-	var from, to *uint64
-
-	getResult := func() (*uint64, *uint64, error) {
-		if from == nil || to == nil {
-			return nil, nil, errors.New("couldn't get any blocks")
-		}
-
-		return from, to, nil
-	}
-
-	var total uint64
+) (uint64, uint64, error) {
```

Comment:
What about using `named parameter` for more readable?

Reply:
done

Reply:
Great job, `e2e` test is a brilliant idea to test backward compatibility. 👍🏻

A little confusion in error handling though: https://github.com/dogechain-lab/dogechain/commit/23da5fc61e644f245d84c8acbda650d7d9abdcf9#r84032903

Reply:
@DarianShawn fast failed and return 5ba4e6f

Reply:
Got it, thanks for the resolved.